### PR TITLE
SC: refactoring mainbridge anchoring data decoding

### PR DESCRIPTION
**Pull request by jeonghonah**
## Proposed changes

- Minor refactoring for duplicated anchoring tx decoding in mainbridge.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jeonghonah** [comment]
@kjhman21 @KimKyungup We decided to rebase `dev-servicechain` with `v1.2.0-rc.1`.

File: blockchain/types/anchoring_data.go [hunk 1]
```diff
@@ -62,3 +67,26 @@ func NewAnchoringDataType0(block *Block, period *big.Int, txCount *big.Int) (*An
 	}
 	return &AnchoringData{AnchoringDataType0, encodedCCTxData}, nil
 }
+
+// DecodeAnchoringTx decodes an anchoring transaction used by main and sub bridges.
+func DecodeAnchoringTx(data []byte) (common.Hash, *big.Int, error) {
```
Comment (aidan-kwon) [inline blockchain/types/anchoring_data.go]: Why don't you use "DecodeAnchroingData"?

File: blockchain/types/anchoring_data.go [hunk 2]
```diff
@@ -62,3 +67,26 @@ func NewAnchoringDataType0(block *Block, period *big.Int, txCount *big.Int) (*An
...
+	if anchoringData.Type == AnchoringDataType0 {
+		anchoringDataInternal := new(AnchoringDataInternalType0)
+		if err := rlp.DecodeBytes(anchoringData.Data, anchoringDataInternal); err != nil {
+			return common.Hash{}, nil, err
+		}
+		logger.Trace("decoded type0 anchoring tx", "blockNum", anchoringDataInternal.BlockNumber.String(), "blockHash", anchoringDataInternal.BlockHash.String(), "txHash", anchoringDataInternal.TxHash.String(), "txCount", anchoringDataInternal.TxCount)
+		return anchoringDataInternal.BlockHash, anchoringDataInternal.BlockNumber, nil
+	} else {
+		return common.Hash{}, nil, errUnknownAnchoringTxType
+	}
```
Comment (aidan-kwon) [inline blockchain/types/anchoring_data.go]: I think we can remove `else` .

File: blockchain/types/anchoring_data.go (hunk 1 above)
Comment (jeonghonah) [reply blockchain/types/anchoring_data.go]: Fixed.

File: blockchain/types/anchoring_data.go (hunk 2 above)
Comment (jeonghonah) [reply blockchain/types/anchoring_data.go]: Fixed.

**Pull request merged**